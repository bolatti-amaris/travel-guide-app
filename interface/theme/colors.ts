type ThemeColors = {
  text: string;
  background: string;
  tint: string;
  icon: string;
  'card-background': string;
  'text-header': string;
  'icon-soft': string;
  'text-dark': string;
  'background-dark': string;
  'tint-dark': string;
  'icon-dark': string;
  'card-background-dark': string;
  'text-header-dark': string;
  'icon-soft-dark': string;
};

const colors: ThemeColors = {
  // Light
  text: '#666666',
  background: '#fafafa',
  tint: '#262626',
  icon: '#262626',
  'card-background': '#ffffff',
  'text-header': '#262626',
  'icon-soft': '#666666',

  // Dark
  'text-dark': '#e6e6e6',
  'background-dark': '#262626',
  'tint-dark': '#e6e6e6',
  'icon-dark': '#e6e6e6',
  'card-background-dark': '#333333',
  'text-header-dark': '#f2f2f2',
  'icon-soft-dark': '#e6e6e6',
};

export default colors;
