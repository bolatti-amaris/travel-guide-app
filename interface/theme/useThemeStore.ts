import { create } from "zustand";
import AsyncStorage from "@react-native-async-storage/async-storage";

type Theme = "light" | "dark";

interface ThemeState {
  theme: Theme;
  setTheme: (theme: Theme) => void;
  loadTheme: () => Promise<void>;
}

export const useThemeStore = create<ThemeState>((set) => ({
  theme: "light",

  setTheme: (theme) => {
    AsyncStorage.setItem("theme", theme);
    set({ theme });
  },

  loadTheme: async () => {
    const saved = await AsyncStorage.getItem("theme");
    if (saved === "light" || saved === "dark") {
      set({ theme: saved });
    }
  },
}));
