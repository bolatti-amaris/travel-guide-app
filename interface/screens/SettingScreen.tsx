import React, { useEffect, useState } from 'react';
import { View, Text, Switch } from 'react-native';
import { useThemeStore } from '@/interface/theme/useThemeStore';

export const SettingsScreen = () => {
  const { theme, setTheme } = useThemeStore();
  const [isDark, setIsDark] = useState(theme === 'dark');

  useEffect(() => {
    setIsDark(theme === 'dark');
  }, [theme]);

  const toggleTheme = () => {
    const newTheme = isDark ? 'light' : 'dark';
    setTheme(newTheme);
    setIsDark(!isDark);
  };

  return (
    <View
      className="flex-1 bg-background dark:bg-background-dark px-4 pt-4 justify-between"
      testID="settings-wrapper"
    >
      <View>
        <View
          className="flex-row justify-between items-center py-4"
          testID="settings-toggle-row"
        >
          <Text
            className="text-xl text-text-header dark:text-text-header-dark font-work-medium"
            testID="settings-label"
          >
            Dark Mode
          </Text>
          <Switch
            value={isDark}
            onValueChange={toggleTheme}
            testID="settings-switch"
          />
        </View>
      </View>

      <View className="items-center pb-6">
        <Text
          className="text-base text-icon dark:text-icon-dark font-work-regular"
          testID="settings-version"
        >
          App Version 0.1.0
        </Text>
      </View>
    </View>
  );
};
