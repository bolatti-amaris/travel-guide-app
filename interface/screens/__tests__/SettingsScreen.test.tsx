import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { SettingsScreen } from '../SettingScreen';

import { useThemeStore } from '@/interface/theme/useThemeStore';

jest.mock('@/interface/theme/useThemeStore', () => ({
  useThemeStore: jest.fn(),
}));

const mockedUseThemeStore = useThemeStore as unknown as jest.Mock;

const mockSetTheme = jest.fn();

describe('SettingsScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders with light theme and matches snapshot', () => {
    mockedUseThemeStore.mockReturnValue({
      theme: 'light',
      setTheme: mockSetTheme,
    });

    const { toJSON } = render(<SettingsScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correct label, switch state and version', () => {
    mockedUseThemeStore.mockReturnValue({
      theme: 'dark',
      setTheme: mockSetTheme,
    });

    const { getByTestId } = render(<SettingsScreen />);

    expect(getByTestId('settings-label').props.children).toBe('Dark Mode');
    expect(getByTestId('settings-switch').props.value).toBe(true);
    expect(getByTestId('settings-version').props.children).toBe(
      'App Version 0.1.0',
    );
  });

  it('toggles the theme on switch interaction', () => {
    mockedUseThemeStore.mockReturnValue({
      theme: 'dark',
      setTheme: mockSetTheme,
    });

    const { getByTestId } = render(<SettingsScreen />);
    fireEvent(getByTestId('settings-switch'), 'valueChange');

    expect(mockSetTheme).toHaveBeenCalledWith('light');
  });

  it('applies correct classNames', () => {
    mockedUseThemeStore.mockReturnValue({
      theme: 'light',
      setTheme: mockSetTheme,
    });

    const { getByTestId } = render(<SettingsScreen />);

    expect(getByTestId('settings-wrapper').props.className).toContain('flex-1');
    expect(getByTestId('settings-label').props.className).toContain(
      'font-work-medium',
    );
    expect(getByTestId('settings-version').props.className).toContain(
      'text-icon',
    );
  });
});
