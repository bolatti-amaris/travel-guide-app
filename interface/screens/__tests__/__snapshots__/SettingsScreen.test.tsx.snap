// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsScreen renders with light theme and matches snapshot 1`] = `
<View
  className="flex-1 bg-background dark:bg-background-dark px-4 pt-4 justify-between"
  testID="settings-wrapper"
>
  <View>
    <View
      className="flex-row justify-between items-center py-4"
      testID="settings-toggle-row"
    >
      <Text
        className="text-xl text-text-header dark:text-text-header-dark font-work-medium"
        testID="settings-label"
      >
        Dark Mode
      </Text>
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "height": 31,
            "width": 51,
          }
        }
        testID="settings-switch"
        value={false}
      />
    </View>
  </View>
  <View
    className="items-center pb-6"
  >
    <Text
      className="text-base text-icon dark:text-icon-dark font-work-regular"
      testID="settings-version"
    >
      App Version 0.1.0
    </Text>
  </View>
</View>
`;
